File: 2016/Python/J3.py
lst = []
word = input()
for i in range(len(word)):
    for x in range(len(word)-1,-1,-1):
        if word[i:x+1]!= "" and word[i:x+1] == word[i:x+1][::-1]:
            lst.append(len(word[i:x+1]))
            break
print(max(lst))

File: 2016/Java/J3.java
import java.util.*;
public class Main {
	public static void main(String args[]) {
        Scanner input = new Scanner(System.in);
        String word = input.next();
        List<Integer> lst = new ArrayList<Integer>();
        for (int i = 0; i<word.length(); i++){
            for (int x = word.length(); x>-1; x--){
                if (word.substring(i,x+1) != "" && word.substring(i,x+1) == new StringBuilder(word.substring(i,x+1)).reverse().toString()){
                    lst.add(Math.abs(x+1-i));
                    break;
                }
            }
        }
        System.out.println((Collections.max(lst)));
    }
}
